Empire/ResearchQueue.cpp

```diff
@@ -323,7 +324,8 @@ void ResearchQueue::insert(iterator it, const std::string& tech_name, bool pause
 { m_queue.insert(it, Element(tech_name, m_empire_id, 0.0f, -1, paused)); }
 
 void ResearchQueue::erase(iterator it) {
-    assert(it != end());
+    if (it == end())
```

Comment:
is this necessarily a crash and/or preferable to throw for, rather than it being a no-op?

Reply:
I have no opinion on this. I implemented the throw considering the previous assert statement. If you think a no-op return would make more sense here, I could change it.